metrics/e2e_test.go

```diff
@@ -0,0 +1,629 @@
+		Name:        "global_export_counts",
+		Description: "Count of exports via standard OpenCensus view.",
+		Measure:     counter,
+		Aggregation: view.Count(),
+	}
+
+	expected := []metricExtract{
+		{"knative.dev/serving/testComponent/global_export_counts", map[string]string{}, 2},
+		{"knative.dev/serving/testComponent/resource_global_export_count", map[string]string{}, 2},
+		{"knative.dev/serving/testComponent/testing/value", map[string]string{"project": "p1", "revision": "r1"}, 0},
+		{"knative.dev/serving/testComponent/testing/value", map[string]string{"project": "p1", "revision": "r2"}, 1},
+	}
+
+	harnesses := []struct {
+		name     string
+		init     func(t *testing.T) error
+		validate func(t *testing.T)
+	}{{
+		name: "Prometheus",
+		init: func(t *testing.T) error {
+			if err := UpdateExporter(context.Background(), configForBackend(prometheus), logtesting.TestLogger(t)); err != nil {
+				return err
+			}
+			// Wait for the webserver to actually start serving metrics
+			return wait.PollImmediate(10*time.Millisecond, 10*time.Second, func() (bool, error) {
+				resp, err := http.Get(fmt.Sprintf("http://localhost:%d/metrics", prometheusPort))
+				return err == nil && resp.StatusCode == http.StatusOK, nil
+			})
+		},
+		validate: func(t *testing.T) {
+			metricstest.EnsureRecorded()
+			resp, err := http.Get(fmt.Sprintf("http://localhost:%d/metrics", prometheusPort))
+			if err != nil {
+				t.Fatalf("failed to fetch prometheus metrics: %+v", err)
+			}
+			defer resp.Body.Close()
+			body, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				t.Fatalf("failed to read prometheus response: %+v", err)
+			}
+			want := `# HELP testComponent_global_export_counts Count of exports via standard OpenCensus view.
+# TYPE testComponent_global_export_counts counter
+testComponent_global_export_counts 2
+# HELP testComponent_resource_global_export_count Count of exports via RegisterResourceView.
+# TYPE testComponent_resource_global_export_count counter
+testComponent_resource_global_export_count 2
+# HELP testComponent_testing_value Test value
+# TYPE testComponent_testing_value gauge
+testComponent_testing_value{project="p1",revision="r1"} 0
+testComponent_testing_value{project="p1",revision="r2"} 1
+`
+			if diff := cmp.Diff(want, string(body)); diff != "" {
+				t.Errorf("Unexpected prometheus output (-want +got):\n%s", diff)
+			}
+		},
+	}, {
+		name: "OpenCensus",
+		init: func(t *testing.T) error {
+			if err := ocFake.start(len(resources) + 1); err != nil {
+				return err
+			}
+			t.Log("Created exporter at", ocFake.address)
+			return UpdateExporter(context.Background(), configForBackend(openCensus), logtesting.TestLogger(t))
+		},
+		validate: func(t *testing.T) {
+			// We unregister the views because this is one of two ways to flush
+			// the internal aggregation buffers; the other is to have the
+			// internal reporting period duration tick, which is at least
+			// [new duration] in the future.
```

Comment:
Are the comments here still correct?

Reply:
Whoops, no. We have a function for that now.